consumer/consumererror/permanent.go

```diff
@@ -14,27 +14,60 @@
 
 package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
 
-import "errors"
+import (
+	"errors"
+	"strconv"
+)
 
-// permanent is an error that will be always returned if its source
+// NO_DATA_COUNT indicates an unknown quantity of telemetry items.
+const NO_DATA_COUNT = -1
+
+// Permanent is an error that will always be returned if its source
 // receives the same inputs.
-type permanent struct {
-	err error
+type Permanent struct {
+	error
+	successful int
+	failed     int
 }
 
-// NewPermanent wraps an error to indicate that it is a permanent error, i.e. an
-// error that will be always returned if its source receives the same inputs.
+// Deprecated [v0.75.0] Use consumererror.NewPermanentWithCounts instead.
 func NewPermanent(err error) error {
-	return permanent{err: err}
+	return Permanent{error: err, successful: NO_DATA_COUNT, failed: NO_DATA_COUNT}
+}
+
+// NewPermanentWithCounts wraps an error to indicate that it is a permanent error, i.e. an
+// error that will be always returned if its source receives the same inputs.
+// The error should also include the number of successful and failed telemetry items
+// if the counts are known. Unknown counts should be set to consumererror.NO_DATA_COUNT.
+func NewPermanentWithCounts(err error, successful, failed int) error {
+	return Permanent{error: err, successful: successful, failed: failed}
```

Comment:
Would you pass `total` when creating this and then derive successful from total - failed?

Reply:
It looks like the `obsreport` package that records the metrics we can use these counts for does this and the total is best calculated by the caller, so I'm leaving this out for now.